File: problems/2913-subarrays-distinct-element-sum-of-squares-i/solution.c
int sumCounts(int* nums, int numsSize) {
    int sum = 0;
    for (int i = 0; i < numsSize; i++) {
        for (int j = i; j < numsSize; j++) {
            int distinctCount = 0;
            int distinct[101] = {0};
            for (int k = i; k <= j; k++) {
                if (distinct[nums[k]] == 0) {
                    distinctCount++;
                    distinct[nums[k]] = 1;
                }
            }
            sum += distinctCount * distinctCount;
        }
    }
    return sum;
}

File: problems/2913-subarrays-distinct-element-sum-of-squares-i/solution.py
class Solution:
    def sumCounts(self, nums: List[int]) -> int:
        n = len(nums)
        ans = 0
        for i in range(n):
            for j in range(i, n):
                sub_array = nums[i:j+1]
                distinct_count = len(set(sub_array))
                ans += distinct_count * distinct_count
        return ans